realign goals, questions, and content (#389)
carpentries-incubator/hpc-intro: _episodes/12-cluster.md

---
title: "Exploring Remote Resources"
teaching: 25
exercises: 10
questions:
- "How does my local computer compare to the remote systems?"
- "How does the login node compare to the compute nodes?"
- "Are all compute nodes alike?"
objectives:
- "Survey system resources using `nproc`, `free`, and the queuing system"
- "Compare & contrast resources on the local machine, login node, and worker
nodes"
- "Learn about the various filesystems on the cluster using `df`"

> - "See `who` else is logged in"

Reviewer: ```suggestion
- "Find out `who` else is logged in"
```

more active than "see"

> - "Glance at the number of idle and occupied nodes"

Reviewer: ```suggestion
- "Assess the number of idle and occupied nodes"
```

As far as I can tell, an exploration of the output of these kinds of monitoring commands is not really written into the episode? Assuming that the Instructor will take time to explain this in a workshop setting, to help learners understand how the output should be parsed, I think "assess" represents the skill they are learning fairly well.

Reviewer 2: In the interests of security, some HPCs are starting to block a user's view of system use, e.g. `qstat`/`squeue` only show the user's own jobs. It may be necessary to tune the lesson to show whatever information/visualisation for the local system may be available. I think the intent here is perhaps along the lines of "Learn about shared system utilisation"?

Author: Right, this episode looks at the output of `sinfo` or `pbsnodes` to see if any nodes are idle. I agree that "assess" is a good descriptor. This is not a job- or user-level inspection, so hopefully skirts the security controls?

Individual computers that compose a cluster are typically called *nodes*
(although you will also hear people call them *servers*, *computers* and
*machines*). On a cluster, there are different types of nodes for different
types of tasks. The node where you are right now is called the *login node*,
*head node*, *landing pad*, or *submit node*. A login node serves as an access
point to the cluster.

As a gateway, the login node should not be used for time-consuming or
resource-intensive tasks. You should be alert to this, and check with your
site's operators or documentation for details of what is and isn't allowed. It
is well suited for uploading and downloading files, setting up software, and

> running tests. Generally speaking, in these lessons, we will avoid running jobs

Reviewer 3: ```suggestion
running short tests that are not resource intensive. Generally speaking, in these lessons, we will avoid running jobs
```

Author: This feels overly prescriptive: the criterion is not run-time or CPU/RAM usage, but rather impact on other users and the ability of the login node to do serve its primary role. As the central point of entry onto a system that may have a different architecture or software stack than a user's off-cluster resources, it is an appropriate place to build software and run test suites, some of which are time-consuming and perform a non-trivial amount of computational work, and that is right and proper.
